### ISSUE TITLE
Value for undeclared variable

### ISSUE THREAD

**Issue by velkovb** [terraform-linters/tflint#2193]
### Summary

We use Terragrunt and have updated it to a version that includes TFLint 0.50.3. This introduced this PR https://github.com/terraform-linters/tflint/pull/1941 which is a breaking change. We do a lot of autogenerated variables that not all modules have and this prevents us from using TFLint.

Would be great if this can be converted into a rule so we can have control over it.

### Command

tflint

### Terraform Configuration

```terraform
terraform {
  before_hook "before_hook" {
    commands = ["validate"]
    execute  = ["tflint"]
  }
}

inputs = {
  account_id   = local.account_vars.account_id,
  account_name = local.account_vars.account_name,
  env          = local.account_vars.env,
  aws_region   = local.region_vars.aws_region,
}

if the module doesn't have any of those defined TFLint won't pass
```

### TFLint Configuration

```HCL
plugin "aws" {
  enabled = true
  version = "0.36.0"
  source  = "github.com/terraform-linters/tflint-ruleset-aws"
}

plugin "terraform" {
  enabled = true
  version = "0.10.0"
  source  = "github.com/terraform-linters/tflint-ruleset-terraform"
}

config {
  call_module_type = "all"
}

rule "terraform_unused_required_providers" {
  enabled = false
}

# We inject a lot of vars that not everyone uses
rule "terraform_unused_declarations" {
  enabled = false
}
```

### Output

```Shell
Error: Value for undeclared variable

  on <value for var.aws_region> line 1:
  (source code not available)

A variable named "aws_region" was assigned, but the root module does not declare a variable of that name.

Error: Value for undeclared variable

  on <value for var.env> line 1:
  (source code not available)
```

### TFLint Version

0.50.3

### Terraform Version

1.9.2

### Operating System

- [X] Linux
- [ ] macOS
- [ ] Windows
